### ISSUE TITLE
Problem with clearml binding into tensorflow's `load_model_from_h5`

### ISSUE THREAD

**Issue by milongo** [allegroai/clearml#355]
Hi,

in `WeightsFileHandler.restore_weights_file` 
(called here: https://github.com/allegroai/clearml/blob/master/clearml/binding/frameworks/tensorflow_bind.py#L1674)

I can see that the argument expected is a string `filepath`. However, if at some point before tensorflow has already loaded up the filepath and built an .h5 object from it, the arguments passed to `PatchKerasModelIO._load_weights`

(https://github.com/allegroai/clearml/blob/master/clearml/binding/frameworks/__init__.py#L34)

will be the tensorflow models instead of the path. Hence it will raise an exception where it complains the arguments are not a filepath.

I don't yet have reproducing code. If the issue above is not clear I can try to make a minimal example.

Basically, consider what happens if this condition evaluates to `True`:
https://github.com/tensorflow/tensorflow/blob/b214a073fbdc9d47a22d6da023cfb0d5316a1747/tensorflow/python/keras/saving/save.py#L200

It will later call this function:

https://github.com/tensorflow/tensorflow/blob/12407f7bde6dfc567cc8eadd5cc56a0810447efe/tensorflow/python/keras/saving/hdf5_format.py#L184

which clearml has a binding to, except that now the model has already been loaded and the arguments to the called function are not filepaths anymore!

**Comment by milongo** [comment]
Reproducing code below:

```import tensorflow as tf
import clearml

from tensorflow.keras.models import load_model as tf_load_model

task = clearml.Task.init(project_name="reproduce bug", task_name="model bug")

mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10)
])



loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)

model.compile(optimizer='adam',
              loss=loss_fn,
              metrics=['accuracy'])

model.fit(x_train, y_train, epochs=1)

model.save("my_model.h5")

saved_model = tf_load_model("my_model.h5")

**Comment by jkhenning** [comment]
Closing this as it was already released. Please reopen if required.
